src/Microsoft.DotNet.CodeAnalysis/PackageFiles/codeAnalysis.targets

```diff
@@ -3,23 +3,53 @@
   <!-- 
     We should be using $(BuildToolsTaskDir) for this, however we are currently forcing that property to 
     always be net45 (because we use Mono to run our tests and Mono needs to load the net45 version)
--->
+  -->
   <PropertyGroup>
     <CodeAnalyzerDir Condition="'$(CodeAnalyzerDir)'=='' and '$(OSEnvironment)'=='Windows_NT'">$(ToolsDir)net45/analyzers/</CodeAnalyzerDir>
     <CodeAnalyzerDir Condition="'$(CodeAnalyzerDir)'=='' and '$(OSEnvironment)'!='Windows_NT'">$(ToolsDir)analyzers/</CodeAnalyzerDir>
   </PropertyGroup>
-  
-  <Import Condition="'$(EnableDotnetAnalyzers)' == 'true'" Project="$(MSBuildThisFileDirectory)PinvokeAnalyzer.props"/>
-	
+
+  <!-- PInvokeChecker data files-->
+  <PropertyGroup Condition="'$(OSGroup)'=='Windows_NT' and '$(EnableDotnetAnalyzers)' == 'true'">
```

Comment:
No I think this should be OSGroup as this isn't about checking only when running on windows but instead only checking the libraries build for windows.